src/DssSpell.t.sol

```diff
@@ -361,7 +361,7 @@ contract DssSpellTest is DssSpellTestBase {
         assertTrue(lerp.done());
     }
 
-    function testNewChainlogValues() public { // don't disable
+    function testNewChainlogValues() private { // make private to disable
```

Comment:
Please follow precedent with regards to this function and revert all changes which alter the behavior of this test, as this is unrelated to the content of this (emergency) spell.

Precedent to follow:
2023-01-11
2023-05-10
2023-05-17

Please reference the procedure for dealing with this [in the checklist](https://github.com/makerdao/pe-checklists/blob/master/spell/spell-crafter-mainnet-workflow.md?plain=1#L26) if there are other concerns. It is my opinion that the objectively menacing sounding comment next to this test suggests that this treatment is necessary for this test, which is clearly an opinion shared by multiple crafters in the past given the precedent.

Reply:
Checklist says nothing specific about comments.

This change was first introduced here: 
https://github.com/makerdao/spells-mainnet/pull/339/files#diff-cd597c47b9a6e2bae57aab767ee9e3d5bb3d8edeb3a2248b4519b9af791cb468R302

This is not a directive to further crafters to not disable this test, it's simply describing what it's doing.

Looking past that, the full history of the archive has the updated comment, which is in line with every other test that can be disabled.

See:
- [2023-01-11](https://github.com/makerdao/spells-mainnet/blob/a3d6cc06ef987689a753e4f1707c20a0b61c82c3/archive/2023-01-11-DssSpell/DssSpell.t.sol#L210)
- [2022-11-12](https://github.com/makerdao/spells-mainnet/blob/a3d6cc06ef987689a753e4f1707c20a0b61c82c3/archive/2022-11-12-DssSpell/DssSpell.t.sol#L136)
- [2022-11-02](https://github.com/makerdao/spells-mainnet/blob/a3d6cc06ef987689a753e4f1707c20a0b61c82c3/archive/2022-11-02-DssSpell/DssSpell.t.sol#L129)

Before that, the practice was to toggle between `// make public to use` and `// make private to disable`, but we abandoned that practice.

See:
- [2022-09-14](https://github.com/makerdao/spells-mainnet/blob/a3d6cc06ef987689a753e4f1707c20a0b61c82c3/archive/2022-09-14-DssSpell/DssSpell.t.sol#L132)

This comment has been misinterpreted and carried over for the past 5 spells.
The test is redundant in all those spells you mentioned. The chainlog version is already checked [here](https://github.com/makerdao/spells-mainnet/blob/a3d6cc06ef987689a753e4f1707c20a0b61c82c3/src/DssSpell.t.base.sol#L1854-L1865).

In reality, the 5 last spells failed to do the proper clean up as per the checklist.

Reply:
An emergency spell is not the context to be introducing test changes where they are not relevant to input or derived actions for that spell.

Test maintenance has historically been done during maintenance PRs or (in cases where the test is blocking certain spell content) test maintenance for a specific test can be done where the spell team unanimously agrees that the test update fits the following criteria.

In this case, this test is not pertinent to the content of the 2023-06-16 spell (which does not contain chainlog changes, new chainlog values or a version bump) and therefore it would generally be considered wise to follow the principle of "if it ain't broke, don't fix it".

The crafter checklist for mainnet spells (which you have not posted) contains pretty clear instructions:
<img width="972" alt="" src="https://github.com/makerdao/spells-mainnet/assets/103920908/ad9e70c5-c77d-4961-b402-65f8a9c95bab">

Simply following this instruction would make the test pass and remove any requirement for making it private. Admittedly the checklist appears to treat all of these options as weighted equally, so this definitely needs clarification in the future. Obviously the 'disable by setting to private' option here clashes with the comment 'don't disable' and so the method to make the test pass (with sufficient coverage) that doesn't involve making it private is clearly what the checklist is instructing you to do.

My personal procedure for spell cleanup is to figure out _what was added by the last spell_ and remove these elements, following archive patterns in order to leave barebones code / comments where necessary. [Removing the chainlog checks for these chainlog keys](https://github.com/makerdao/spells-goerli/pull/202/files#diff-cd597c47b9a6e2bae57aab767ee9e3d5bb3d8edeb3a2248b4519b9af791cb468R364) and leaving the version check for 1.14.13 will make the test pass.

I prioritize security and if a test says `don't disable` I'm not going to approve a hastily made change to it within a 24 hour SLA emergency spell, regardless of whether this comment makes logical sense or not.

As the crafter for this spell, your role is to ship the instruction set provided by Governance Facilitators as spell code. Perhaps you are correct, but harking back to the first line, this is not the context for this change as it is not at all relevant to the current spell's content.

All that debating about this is doing is delaying the shipping of an emergency spell over an extraneous (functional) change which I am not comfortable with shipping given our current context. Note that my objection to this change _does not block spell content_ nor does it affect any test coverage related to the spell content and is rooted in a strong prioritization of security.

Reply:
It's not a functional change, it's a comment. More important than that, it's a wrong comment.

This is literally an empty spell, I've shown you why the comment is wrong, I don't get why you need to make a fuss about it.

Don't be [the boy who cried wolf](https://en.wikipedia.org/wiki/The_Boy_Who_Cried_Wolf), otherwise people will not take you seriously when something is actually wrong.

Reply:
I have no objection to fixing the code in the future (in fact, your points as outlined above make sense and I would support making an update to our test setup for chainlog checks). In fact, quite a lot of our tests need updating, maintenance and hardening.

However, describing _disabling_ a test marked `don't disable` as "not a functional change" is drastically incompetent at best and actively malicious at worst. You are _disabling_ a _function_. At best, this is functionally equivalent due to coverage elsewhere, but at worst this leads to a lack of coverage from this spell onwards until someone catches it in the future.

Either way, this is not something I am comfortable shipping with such short notice given that it has nothing to do with the current spell's content, so I cannot resolve this issue while a _functional_ change that I disagree with is present, even if it's just checking the chainlog version. I am not approving this in the same way that I would not approve disabling `testPSMs` (which is another test with a note on it to be left as enabled / public) in a emergency spell. It's not a matter of whether you're right or wrong and more a matter of this being the wrong place and time for this kind of change.

Reply:
> However, describing disabling a test marked `don't disable` as "not a functional change" is drastically incompetent at best and actively malicious at worst.

You are joking, right? This must be a joke. Where are the cameras?

> You are disabling a function. At best, this is functionally equivalent due to coverage elsewhere, but at worst this leads to a lack of coverage from this spell onwards until someone catches it in the future.

And removing the content of the test is what besides disabling it?

The only thing present in the test now is checking the chainlog version, [which is already done in other test](https://github.com/makerdao/spells-mainnet/blob/2613db457ed4f8ea4edf92f1d3ca4c0ca7684f26/src/DssSpell.t.base.sol#L1864).

If you spent half of the energy you spent writing these comments simply going through what I highlighted, we wouldn't be having this discussion.

Reply:
> And removing the content of the test is what besides disabling it?

Precedent leaves version check, see archives, it isn't empty.

> [which is already done in other test](https://github.com/makerdao/spells-mainnet/blob/2613db457ed4f8ea4edf92f1d3ca4c0ca7684f26/src/DssSpell.t.base.sol#L1864).

Yes, this is probably true. That's why this gets fixed when _spell tests are being maintained or upgraded_ and _not in an emergency spell_, since this has nothing to do with declaring a DAO Resolution.

Reply:
> Precedent leaves version check, see archives, it isn't empty.

This is false. There are only 2 instances in which this was done like that:

[2023-05-10](https://github.com/makerdao/spells-mainnet/blob/f5aba05ae068742bc7a9352069d58f317d41691d/archive/2023-05-10-DssSpell/DssSpell.t.sol#L239-L245)
[2023-05-17](https://github.com/makerdao/spells-mainnet/blob/f5aba05ae068742bc7a9352069d58f317d41691d/archive/2023-05-17-DssSpell/DssSpell.t.sol#L266-L272)

Every single other spell in the archive which contains this test is either active (meaning there are new chainlog entries to test) or marked private.